regression-test/suites/correctness_p0/test_like_predicate_with_concat.groovy

```diff
@@ -0,0 +1,90 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_like_predicate_with_concat") {
+    sql """ DROP TABLE IF EXISTS `test_like_predicate_with_concat` """
+
+    sql """
+        CREATE TABLE IF NOT EXISTS `test_like_predicate_with_concat` (
+        `id` int,
+        `value_col` string,
+        `pattern_col` string
+        ) ENGINE=OLAP
+        DISTRIBUTED BY HASH(`id`) BUCKETS 4
+        PROPERTIES (
+        "replication_allocation" = "tag.location.default: 1"
+        );
+    """
+
+    sql """ 
+        INSERT INTO `test_like_predicate_with_concat` VALUES 
+            (0, 'prefix0_infix0_suffix0', 'prefix0'),
```

Comment:
The test is too simple. I think you should add some special chars such as $ ^ % in the col value. And it could at the beginning of the value or ending of the value or middle of the value.

Reply:
Okay, I will refine this test case.